components/scream/scripts/test_all_scream.py

```diff
@@ -187,11 +194,11 @@ def __init__(self, cxx_compiler=None, f90_compiler=None, c_compiler=None,
         if self._parallel:
             # We need to be aware that other builds may be running too.
             # (Do not oversubscribe the machine)
-            make_physical_cores_per_test = get_available_cpu_count(logical=False) // len(self._tests)
             log_per_phys = logical_cores_per_physical_core()
 
-            make_jobs_per_test  = make_physical_cores_per_test * log_per_phys
-            ctest_jobs_per_test = ctest_max_jobs // len(self._tests)
+            # Avoid splitting physical cores across test types
+            make_jobs_per_test  = ((make_max_jobs // log_per_phys) * log_per_phys) // len(self._tests)
```

Comment:
I'm not sure what's the point of dividing-multiply by log_per_phys. All it does is making sure that the result is divisible by log_per_phys, but that's not a real issue, right? And when we divide by `len(self._tests)` we can still end up with a physical core split across builds. Perhaps you wanted to do this:
```
ctest_jobs_per_test = ( (ctest_max_jobs // log_per_phys) // len(self._tests) ) * log_per_phys
```
so that you split physical cores across builds, then add back the hw firepower.

Reply:
@bartgol , there was a bug here and it looks like you may have reviewed the outdated code.